### PR TITLE
Miscellaneous test changes to avoid making assumptions about what's a valid URL

### PR DESCRIPTION
**Pull request by d-ronnqvist**
Bug/issue #, if applicable: rdar://108237260

## Summary

This makes a handful of changes, mainly in tests, to avoid making assumptions about what's a valid URL. Each change is its own commit that describe the reason for that change.

> Note: It also makes one behavioral change by formalizing the requirement that source repository base URLs specify both a scheme and a host. 

I _think_ that's the type of values we expect for this configuration. If we don't have any specific validation for these values I think we should remove those test assertions (and possibly use a string instead of a URL)

## Dependencies

None

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [ ] Added tests
- [ ] Ran the `./bin/test` script and it succeeded
- [ ] Updated documentation if necessary

### CODE REVIEW

**Review by franklinsch — APPROVED** [review]

**Comment by d-ronnqvist** [comment]
@swift-ci please test

**Comment by d-ronnqvist** [comment]
@swift-ci please test

File: Tests/SwiftDocCTests/Infrastructure/ResolvedTopicReferenceTests.swift [hunk 1]
```diff
@@ -54,7 +54,7 @@ class ResolvedTopicReferenceTests: XCTestCase {
         do {
             let resolvedOriginal = ResolvedTopicReference(bundleIdentifier: "bundleID", path: "/path/sub-path", fragment: "fragment", sourceLanguage: .swift)
             
-            let unresolved = UnresolvedTopicReference(topicURL: ValidatedURL(parsingExact: "doc://host.name/;;;")!)
+            let unresolved = UnresolvedTopicReference(topicURL: ValidatedURL(parsingExact: "doc://host.name/   ")!)
```
Comment (QuietMisdreavus) [inline Tests/SwiftDocCTests/Infrastructure/ResolvedTopicReferenceTests.swift:57]:
```
Test Suite 'Selected tests' started at 2023-07-25 18:53:24.131
Test Suite 'SwiftDocCPackageTests.xctest' started at 2023-07-25 18:53:24.132
Test Suite 'ResolvedTopicReferenceTests' started at 2023-07-25 18:53:24.132
Test Case '-[SwiftDocCTests.ResolvedTopicReferenceTests testAppendingReferenceWithEmptyPath]' started.
SwiftDocCTests/ResolvedTopicReferenceTests.swift:57: Fatal error: Unexpectedly found nil while unwrapping an Optional value
```

Looks like this change made this URL initializer fail in CI.

**Comment by d-ronnqvist** [comment]
@swift-ci please test

**Comment by d-ronnqvist** [comment]
@swift-ci please test

**Comment by d-ronnqvist** [comment]
@swift-ci please test

**Pull request merged**